cmd/infra/aws/iam.go

```diff
@@ -682,3 +667,60 @@ func iamTags(infraID, name string) []*iam.Tag {
 	}
 	return tags
 }
+
+type oidcTrustPolicyParams struct {
+	ProviderARN     string
+	ProviderName    string
+	ServiceAccounts string
+}
+
+const (
+	oidcTrustPolicyTemplate = `{
+		"Version": "2012-10-17",
+		"Statement": [
+			{
+				"Effect": "Allow",
+				"Principal": {
+					"Federated": "{{ .ProviderARN }}"
+				},
+					"Action": "sts:AssumeRoleWithWebIdentity",
+				"Condition": {
+					"StringEquals": {
+						"{{ .ProviderName }}:sub": {{ .ServiceAccounts }}
+					}
+				}
+			}
+		]
+	}`
+)
+
+func oidcTrustPolicy(providerARN, providerName string, serviceAccounts ...string) string {
+	params := oidcTrustPolicyParams{
+		ProviderARN:  providerARN,
+		ProviderName: providerName,
+	}
+	if len(serviceAccounts) == 1 {
+		params.ServiceAccounts = fmt.Sprintf("%q", serviceAccounts[0])
+	} else {
+		sas := &bytes.Buffer{}
+		fmt.Fprintf(sas, "[")
+		for i, sa := range serviceAccounts {
+			fmt.Fprintf(sas, "%q", sa)
+			if i < len(serviceAccounts)-1 {
+				fmt.Fprintf(sas, ", ")
```

Comment:
@csrwng just making sure you tested this.  I've never tried to put two SAs as the `sub`.